apis/v1alpha3/backendtlspolicy_types.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories=gateway-api,shortName=btlspolicy
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+//
+// BackendTLSPolicy is a Direct Attached Policy.
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=Direct"
+
+// BackendTLSPolicy provides a way to configure how a Gateway
+// connects to a Backend via TLS.
+type BackendTLSPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of BackendTLSPolicy.
+	Spec BackendTLSPolicySpec `json:"spec"`
+
+	// Status defines the current state of BackendTLSPolicy.
+	Status PolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// BackendTLSPolicyList contains a list of BackendTLSPolicies
+type BackendTLSPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BackendTLSPolicy `json:"items"`
+}
+
+// BackendTLSPolicySpec defines the desired state of BackendTLSPolicy.
+//
+// Support: Extended
+type BackendTLSPolicySpec struct {
+	// TargetRef identifies an API object to apply the policy to.
+	// Only Services have Extended support. Implementations MAY support
+	// additional objects, with Implementation Specific support.
+	// Note that this config applies to the entire referenced resource
+	// by default, but this default may change in the future to provide
+	// a more granular application of the policy.
+	//
+	// Support: Extended for Kubernetes Service
+	//
+	// Support: Implementation-specific for any other resource
+	//
+	TargetRef PolicyTargetReferenceWithSectionName `json:"targetRef"`
```

Comment:
Given the overall painfulness of breaking changes, we _may_ want to get ahead of one of the ones that seems very likely, plural targetRefs. This was one of the takeaways from our policy-focused meeting + KubeCon discussions that seemed to have broad consensus. More discussion in https://github.com/kubernetes-sigs/gateway-api/discussions/2927#discussioncomment-8991869. (Need more discussion before actually committing to a change here, but want to raise this as a discussion point so we can potentially avoid another breaking change in v1.2).

/cc @arkodg @gcs278 @guicassolato @mikemorris @shaneutt

Reply:
I don't feel strongly about this, but wanted to point out that in GEP-1897 we wanted BackendTLSPolicy to solve the problem of how to indicate whether a connection needs TLS certificate validation.  We proposed a solution in which we watch connections to the TargetRef and determine if it exists in a  BackendTLSPolicy.  Having multiple TargetRefs per policy could introduce issues with that so we should propose how to do this efficiently.
We also don't allow wildcard Hostname in the Validation.

> This solution proposes, as introduced in [GEP-713](https://gateway-api.sigs.k8s.io/geps/gep-713/), that the implementation should watch the connections to a specified TargetRef (Service), and if the Service matches a BackendTLSPolicy, then assume the connection is TLS, and verify that the TargetRef’s certificate can be validated by the client (Gateway) using the provided certificates and hostname before the connection is made. 

However, we do have something else listed as a Graduation Criterion in GEP-1897: 

> Rob Scott is interested in extending the TargetRef to optionally include port, since we are targeting the entirety
> of a Service. See the discussion in https://github.com/kubernetes-sigs/gateway-api/pull/2113/files#r1231594914,
> and follow up issue in https://github.com/kubernetes-sigs/gateway-api/issues/2147

Reply:
I described an example scenario over in https://github.com/kubernetes-sigs/gateway-api/discussions/2927#discussioncomment-9023920 for why plural targetRefs is specifically useful for BackendTLSPolicy during blue/green migrations, would be a big +1 to this, but I can handle it in a subsequent PR if we'd prefer to avoid blocking this on that slightly-wider change.

Reply:
x-ref #2966

Reply:
All set.